torchgen/gen.py

```diff
@@ -2281,6 +2282,24 @@ def gen_op_headers(
             )
         },
     )
+    view_copy_with_symint_pairs: List[Tuple[NativeFunction, NativeFunction]] = []
```

Comment:
Thanks for asking - this is part of the "race" where more SymInt support to LTC was getting added while I was landing this PR, so I added just enough SymInt support for functionalization to get the tests to pass.

The functionalization codegen has existing logic to generate kernels for all view_copy ops, that basically just call the underlying view op and then perform a clone.

That's no longer what we want for the SymInt overloads, though (cc @Krovatkin): we have an existing implementation for `narrow_copy.SymInt` ([link](https://github.com/pytorch/pytorch/blob/817eb94ff494e07e94a9effb626ce2121f226a54/aten/src/ATen/native/TensorShape.cpp#L930)), that just "concretizes" the sym ints and calls into the normal `view_copy` kernel.

Since we need `SymInt` overloads of all of our view copy operators anyway, I added codegen here to automatically generate those kernels for all `view_copy.SymInt` overloads: they will just call `.expectInt()` or `.toIntArrayRefSlow()` on all the sym int inputs, and call the non-symint overload.

To do that though, the codegen needs to know how to pair up the `view_copy` and `view_copy.SymInt` `NativeFunction` objects. That pairing up is what this code is doing.

I remember hearing from our dynamic shapes chat that in the end state we'll probably end up with just having a single `SymInt` overload for these ops instead of having both the symbolic + concrete overloads. I think that it'll be pretty easy to remove this codegen if/once we get there.